### PR TITLE
Refactor code to avoid use of rich library

### PR DESCRIPTION
**Pull request by ssbarnea**
Related: https://issues.redhat.com/browse/AAP-36125

### CODE REVIEW

**Review by abhikdps — APPROVED** [review]

File: src/ansiblelint/output.py [hunk 1]
```diff
@@ -43,63 +42,367 @@
...
+            "repr.path": (style.path, style.normal),
+            "repr.number": (style.number, style.normal),
+            "repr.link": (style.link, style.normal),
+            "failed": (style.failed, style.normal),
+            "success": (style.success, style.normal),
+        }
+        # Regex to find opening tags and their content
+        tag_pattern = re.compile(r"\[([\w\.]+)(?:=(.*?))?\]|\[/\]")
+
+        def replace_bb_links(text: str) -> str:
```
Comment (cidrblock) [inline src/ansiblelint/output.py:304]: Can you get rid of these nested functions?  It's really hard to grok.

File: src/ansiblelint/output.py [hunk 2]
```diff
@@ -43,63 +42,367 @@
...
+
+            Returns:
+                str: The text with BBCode links replaced by HTML <a> tags.
+            """
+            # Define a safe regex pattern
+            pattern = r"\[link=([^\]]+)\]((?:[^\[]|\[(?!\/link\]))+)\[/link\]"
+
+            # Replace matches with HTML <a> tags
+
+            def replacement(match: re.Match[str]) -> str:
```
Comment (cidrblock) [inline src/ansiblelint/output.py:318]: 3 deep?

File: src/ansiblelint/output.py [hunk 3]
```diff
@@ -43,63 +42,367 @@
...
+    ) -> None:
+        """Internal print implementation."""
+        buffer = StringIO()
+        print(*values, sep=sep, end="", file=buffer, flush=True)
+        buffer.seek(0)
+        data = buffer.read()
+        print(self.render(data), end=end, file=file or self._file, flush=flush)
+
+    def render(self, text: str) -> str:
+        """Parses a string containing nested BBCode with a generic block terminator ([/])."""
```
Comment (cidrblock) [inline src/ansiblelint/output.py:282]: Please add params to docstrings in google format :)

File: src/ansiblelint/output.py [hunk 4]
```diff
@@ -43,63 +42,367 @@
...
+    def render_link(cls, uri: str, label: str | None = None) -> str:
+        """Return a link."""
+        return label or uri
+
+
+@dataclasses.dataclass
+class AnsiStyle(PlainStyle):
+    """Theme."""
+
+    @dataclass
```
Comment (cidrblock) [inline src/ansiblelint/output.py:163]: This is atypical, was there a need to nest this?

File: src/ansiblelint/output.py [hunk 5]
```diff
@@ -43,63 +42,367 @@
...
+
+    def print(
+        self,
+        *values: Any,
+        sep: str | None = " ",
+        end: str | None = "\n",
+        file: TextIO | None = None,
+        flush: bool = False,
+    ) -> None:
+        """Internal print implementation."""
```
Comment (cidrblock) [inline src/ansiblelint/output.py:274]: Can you comment why this was necessary on the docstring?

File: src/ansiblelint/output.py [hunk 6]
```diff
@@ -43,63 +42,367 @@
...
+            "noteset": (style.notset, style.normal),
+            # data types
+            "repr.path": (style.path, style.normal),
+            "repr.number": (style.number, style.normal),
+            "repr.link": (style.link, style.normal),
+            "failed": (style.failed, style.normal),
+            "success": (style.success, style.normal),
+        }
+        # Regex to find opening tags and their content
+        tag_pattern = re.compile(r"\[([\w\.]+)(?:=(.*?))?\]|\[/\]")
```
Comment (cidrblock) [inline src/ansiblelint/output.py:302]: I've found putting the regex and the use of in it's own fn to make testing easier and demostrate what it does in the unit tests.

File: src/ansiblelint/output.py [hunk 7]
```diff
@@ -43,63 +42,367 @@
...
+        BRIGHT_MAGENTA = "\033[95m"
+        BRIGHT_CYAN = "\033[96m"
+        BRIGHT_WHITE = "\033[97m"
+        END = "\033[0m"
+        # more complex
+        BOLD = "\033[1m"
+        DIM = "\033[1;30m"
+        BOLD_CYAN = "\033[1;36m"
+
+    warning = "\033[33m"  # yellow
```
Comment (cidrblock) [inline src/ansiblelint/output.py:189]: Can this ref the constant in ANSI?

File: src/ansiblelint/output.py [hunk 8]
```diff
@@ -43,63 +42,367 @@
...
+    error = ANSI.RED  # "\033[31m"  # red
```
Comment (cidrblock) [inline src/ansiblelint/output.py:190]: Comment here probably not needed

**Review by cidrblock — COMMENTED** [review]
Some nitpicky stuff, looks great tho.  Sorry about the late review,

**Pull request merged**